src/vm/internal/otable.cc

```diff
@@ -15,257 +21,236 @@
  * Note: if you change an object name, you must remove it and reenter it.
  */
 
-static int otable_size;
-static int otable_size_minus_one;
 
-static object_t *find_obj_n(const char * /*s*/);
-
-/*
- * Object hash function, ripped off from stralloc.c.
- */
-#define ObjHash(s) whashstr(s) & otable_size_minus_one
-
-/*
- * hash table - list of pointers to heads of object chains.
- * Each object in chain has a pointer, next_hash, to the next object.
- */
-
-static object_t **obj_table = 0;
-static object_t **ch_table = 0;
-
-static char *_basename(const char *full, int *size) {
-  char *tmp;
-  char *name = new_string(*size = strlen(full), "base_name: name");
-  while (*full == '/') {
-    full++;
-    (*size)--;
-  }
-  strcpy(name, full);
-  tmp = strchr(name, '#');
-  if (tmp) {
-    *size = tmp - name;
-  }
-  while (*size > 2 && name[*size - 1] == 'c' && name[*size - 2] == '.') {
-    *size -= 2;
-  }
-  name[*size] = 0;
-  name = extend_string(name, *size);
-  return name;
-}
+std::unordered_map< std::string, struct object_t* > ObjectTable::objects_;
```

Comment:
Static class variables reside within the data segment and as such when you are using them  they are already initialised according the source given value or, in case of classes, the given constructor ...
That means also that there can´t be any true leak for such variables, leaks can only happen with heap allocated variables, stack aallocated variables automatically freed when leaving the scope in which they are defined ...